GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        public T Value { get; }
+
+        public Node<T> Next { get; private set; }
+
+        public Node(T value)
+        {
+            Value = value;
+            Next = this;
+        }
+
+        public Node<T> Append(T value)
+        {
+            if (Exists(value))
+                throw new ArgumentException("Cannot add same value more than once", nameof(value));
```

Comment:
```suggestion
            if (Exists(value)) throw new ArgumentException("Cannot add same value more than once", nameof(value));
```
This is a minor nit pick, but I'd suggest either keeping this inline or using curly brackets once you start to multi-line it.